### ISSUE TITLE
Correct way to get client port?

### ISSUE THREAD

**Issue by MichaelLeeHobbs** [mscdex/ssh2#677]
On connection currently getting it this way: client._sock._peername.port is this the only way?

**Comment by mscdex** [comment]
If anything you should use `socket.remotePort` instead of a node core underscore-prefixed property. However, you shouldn't really need to access `_sock` either. This could cause problems in case someone passes a stream in as the underlying "socket", which may not have `remote*` properties.

**Comment by MichaelLeeHobbs** [comment]
Thanks for the reply. Right, don't want to use an underscore-prefixed property but not seeing a way on server connection callback. This is how I am setting up the server:

    start() {
        let {ip, port} = config.listen  // todo add test for bad config values
        new ssh2.Server({
            hostKeys: [fs.readFileSync('id_rsa')]  // todo add to config
        }, (client, info) => this._handleConnection(client, info)).listen(port, ip, () => {
            this.logger('-', '-', `Listening on ${ip}:${port}`)
        })
    }

In the `_handleConnection` function I am setting up a `clientData` object like so.

        let now = Date.now()
        let clientData = {
            id: `${info.ip}:${client._sock._peername.port || now}`,
            ip: info.ip,
            port: client._sock._peername.port || now,
            identRaw: info.identRaw,
            username: '-',
            authMethod: '-'
        }
        this.logger(clientData.id, clientData.username, `${new Date().toISOString()} "connected"`)

From there on out, every action is logged using the clientData.id. Thus, in the `client` or `info` objects not seeing a clean way to grab it. I'm hoping I'm missing something but if not this will have to do as this is for a HIPPA project so logging is almost more important than functionality, lol.

Thanks,
Michael

**Comment by MichaelLeeHobbs** [comment]
Yeah, I'm actually writing a wrapper sever to wrap an old unsecure system from the early 90's, fun stuff.
